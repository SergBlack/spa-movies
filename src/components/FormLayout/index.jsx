import React from 'react';
import { node, string } from 'prop-types';
import styled from 'styled-components';

import Title from '@components/Title';

const StyledForm = styled.div`
  display: flex;
  flex-direction: column;
  width: 80%;
`;

const TitleWrapper = styled.div`
  display: flex;
  margin: 20px 0;
`;

const FormLayout = ({ title, children }) => (
  <StyledForm>
    <TitleWrapper>
      <Title content={title} size="30px" uppercase />
    </TitleWrapper>
    {children}
  </StyledForm>
);

FormLayout.propTypes = {
  title: string,
  children: node.isRequired,
};

FormLayout.defaultProps = {
  title: '',
};

export default FormLayout;
