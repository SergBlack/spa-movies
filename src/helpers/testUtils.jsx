/* eslint-disable react/prop-types */
import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '../App';
import rootReducer from '../redux/rootReducer';

function render(
  ui,
  {
    initialState,
    store = createStore(rootReducer, initialState),
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return (
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          {children}
        </Provider>
      </ThemeProvider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
