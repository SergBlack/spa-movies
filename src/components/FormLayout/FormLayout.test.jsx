import React from 'react';
import { render } from '@helpers/testUtils';

import FormLayout from './index';

const SomeComponent = () => <div>Some content</div>;

describe('FormLayout', () => {
  test('renders FormLayout snapshot', () => {
    const { asFragment } = render(
      <FormLayout title="FormLayout title">
        <SomeComponent />
      </FormLayout>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
