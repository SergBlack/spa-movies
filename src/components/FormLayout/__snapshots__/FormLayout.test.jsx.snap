// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLayout renders FormLayout snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsTCUz gVOTKC"
  >
    <div
      class="sc-dlfnbm dOXLOI"
    >
      <h1
        class="sc-bdfBwQ dDltp"
      >
        FormLayout title
      </h1>
    </div>
    <div>
      Some content
    </div>
  </div>
</DocumentFragment>
`;
